relay-common/src/glob3.rs

```diff
@@ -12,6 +12,50 @@ fn is_match(globs: &[Regex], message: &[u8]) -> bool {
     globs.iter().any(|regex| regex.is_match(message.as_ref()))
 }
 
+/// A singular glob-pattern.
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct GlobPattern(GlobPatterns);
+
+impl GlobPattern {
+    /// Returns a new instance of [`GlobPattern`].
+    pub fn new(pattern: impl Into<String>) -> Self {
```

Comment:
```suggestion
    pub fn new(pattern: String) -> Self {
```

We usually don't do this.

Personally I would use this only for builders, it is otherwise barely useful (usually only in tests) and hides accidental allocations (passing `&foo` instead of `foo` by accident).